src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -636,6 +649,8 @@ public bool DontFragment
         {
             get
             {
+                if (OperatingSystem.IsWasi()) throw new PlatformNotSupportedException(); // TODO remove with https://github.com/dotnet/runtime/pull/107185
```

Comment:
would it make sense to return `false` like the `MulticastLoopback` bellow? In general, I feel throwing on getters is somewhat unpleasant.

Reply:
done